### PR TITLE
github: Migrate Solaris Check To Ubuntu

### LINKED ISSUE #3459: Github Action For Solaris 10 Build Failing
The Github action that tests a build for Solaris 10 using Vagrant is failing with the following error , I believe this is due to issues with the underlying GitHub Runner.

```
    adoptopenjdkSol10: SSH username: vagrant
    adoptopenjdkSol10: SSH auth method: private key
The guest machine entered an invalid state while waiting for it
to boot. Valid states are 'starting, running'. The machine is in the
'gurumeditation' state. Please verify everything is configured
properly and try again.

If the provider you're using has a GUI that comes with it,
it is often helpful to open that and watch the machine, since the
GUI often has more helpful error messages than Vagrant can retrieve.
For example, if you're using VirtualBox, run `vagrant up` while the
VirtualBox GUI is open.

The primary issue for this error is that the provider you're using
is not properly configured. This is very rarely a Vagrant issue.
Error: Process completed with exit code 1.
```

### PR DESCRIPTION
**Pull request by steelhead31**
Fixes #3459 

Migrates the failing Solaris 10 build action from Macos 12 to Ubuntu Latest & removes the insecure flag from the ssh connection to the vagrant box..

`-o StrictHostKeyChecking=no`

##### Checklist

- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [X] VPC/QPC not applicable for this PR

### CODE REVIEW

File: .github/workflows/build_vagrant.yml [hunk 1]
```diff
@@ -19,54 +19,73 @@ permissions:
...
-        [ ! -r $HOME/.ssh/known_hosts ] && touch $HOME/.ssh/known_hosts && chmod 644 $HOME/.ssh/known_hosts
-        ssh-keygen -R $(cat playbooks/AdoptOpenJDK_Unix_Playbook/hosts.unx)
-        sed -i -e "s/.*hosts:.*/  hosts: all/g" playbooks/AdoptOpenJDK_Unix_Playbook/main.yml
-        awk '{print}/^\[defaults\]$/{print "private_key_file = id_rsa"; print "timeout = 60"; print "remote_tmp = $HOME/.ansible/tmp"}' < ansible.cfg > ansible.cfg.tmp && mv ansible.cfg.tmp ansible.cfg
+        sudo apt-get install vagrant
 
-    - name: Run Ansible Playbook
+    - name: Cache Solaris10.box.gz
+      id: solaris-10-cache
+      uses: actions/cache@v4
```
Comment (gdams) [inline .github/workflows/build_vagrant.yml:41]:
```suggestion
      uses: actions/cache@ab5e6d0c87105b4c9c2047343972218f562e4319 # v4.0.1
```

File: .github/workflows/build_vagrant.yml [hunk 2]
```diff
@@ -19,54 +19,74 @@ permissions:
...
+      with:
+        path: Solaris10.box.gz
+        key: sol10box
+
+    - name: Download Solaris 10 Box If Cache Misses
+      if: steps.cache.outputs.cache-hit != 'true'
```
Comment (gdams) [inline .github/workflows/build_vagrant.yml:47]:
```suggestion
      if: steps.solaris-10-cache.outputs.cache-hit != 'true'
```

File: .github/workflows/build_vagrant.yml [hunk 3]
```diff
@@ -19,36 +19,57 @@ permissions:
...
-          else
-            echo "Sum Bad"
-            exit 99;
-          fi
+          echo "Sum Bad"
+          exit 99;
         fi
 
+    - name: Add Solaris 10 Box To Vagrant From Cache
+      if: steps.solaris-10-cache.outputs.cache-hit == 'true'
```
Comment (gdams) [inline .github/workflows/build_vagrant.yml:67]: Guessing this will need to happen each time regardless of cache?

Comment (steelhead31) [reply .github/workflows/build_vagrant.yml:67]: Yes, its the process of adding the VM to Vagrant, as its installed on each machine... on our permanent Vagrant hosts, this is a one off process, but GHA runners don't persist.

**Review by gdams — APPROVED** [review]

**Review by andrew-m-leonard — APPROVED** [review]
lgtm

**Pull request merged**